app/javascript/components/ProfileForm.jsx

```diff
@@ -0,0 +1,229 @@
+        <Input defaultValue={this.state.prevValues[index]} onChange={(e) => this.handleChange(index, e)} />
+      </div>
+    )
+  }
+
+  renderTextarea(name, index) {
+    const { TextArea } = Input;
+    return (
+      <div key={index}>
+        <label>{this.state.niceFieldNames[index]}</label>
+        <TextArea defaultValue={this.state.prevValues[index]} rows ={4} onChange={(e) => this.handleChange(index, e)} autosize={true}/>
+      </div>
+    )
+  }
+
+  renderDatePicker(name, index) {
+    if (this.state.mode == "create") {
+      this.state.prevValues[index] = moment().format("YYYY-MM-DD")
+    }
+    return (
+      <div key={index}>
+        <label>{this.state.niceFieldNames[index]}</label>
+        <DatePicker defaultValue={this.state.prevValues[index] == "" ? moment() : moment(this.state.prevValues[index])} 
+                    onChange={(date, dateString) => this.state.prevValues[index] = dateString} />
+      </div>
+    )
+  }
+
+  renderDropdown(name, index) {
+    const Option = Select.Option;
+    return (
+      <div key={index}>
+        <label>{this.state.niceFieldNames[index]}</label>
+        <Select defaultValue={this.state.prevValues[index]} style={{width: 220}} onChange={(e) => this.state.prevValues[index] = e}>
+          {this.state.fieldTypes[index].map(option => 
+            <Option key={index} value={option}>{option}</Option>)}
+        </Select>
+      </div>
+    )
+  }
+
+  renderSlider(name, index) {
+    return (
+      <div key={index}>
+        <label>{this.state.niceFieldNames[index]} - {this.state.niceFieldNames[index+1]}</label>
+        <Slider range defaultValue={[this.state.prevValues[index], this.state.prevValues[index+1]]}
+                onChange={(e) => [this.state.prevValues[index], this.state.prevValues[index+1]] = [e[0], e[1]]}
+                max={5000}/>
+      </div>
+    );
+  }
+
+  renderForm() {
+    return (
+      this.state.fieldNames.map((name, index) => {
+        if (this.state.fieldTypes[index] === "textbox") {
+          return ( 
+            this.renderTextbox(name, index)
+          )
+        } else if (this.state.fieldTypes[index] === "textarea") {
+          return (
+            this.renderTextarea(name, index)
+          )
+        } else if (this.state.fieldTypes[index] === "datepicker") {
+          return (
+            this.renderDatePicker(name, index)
+          )
+        } else if (this.state.fieldTypes[index] === "slider") {
+          return (
+            this.renderSlider(name, index)
+          )
+        } else if (this.state.fieldTypes[index] === "_slider") {
+          return null
+        } else {
+          return (
+            this.renderDropdown(name, index)
+          )
+        }
+      })
+    )}
+
+  render() {
+    let returnArr = [];
+    if (this.state.mode === "create") {
+      returnArr = [this.renderForm(),
+          <Button type="primary" onClick={this.handleCreate}>Submit</Button>,
```

Comment:
try adding a key to all of the buttons that is unique from all other keys since they are all within an array

Reply:
yup this did the trick